### PR TITLE
fix: Replace wildcards in RBAC objects with explicit resources and verbs

### PR DESCRIPTION
**Pull request by legal90**
According to Kubernetes documentation and various k8s security guidelines, wildcards in resource and verb entries should be avoided:

> [!WARNING]
> Using wildcards in resource and verb entries could result in overly permissive access being granted to sensitive resources. For instance, if a new resource type is added, or a new subresource is added, or a new custom verb is checked, the wildcard entry automatically grants access, which may be undesirable. The [principle of least privilege](https://kubernetes.io/docs/concepts/security/rbac-good-practices/#least-privilege) should be employed, using specific resources and verbs to ensure only the permissions required for the workload to function correctly are applied.

Refs: 
- https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding
- https://cloud.google.com/kubernetes-engine/docs/best-practices/rbac#no-wildcards

This PR could be seen as a continuation of a previous work for hardening the RBAC: https://github.com/kedacore/charts/pull/625
It replaces `*` with explicit verbs and resources, according to KEDA needs.

### Checklist

- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update our Helm chart: https://github.com/kedacore/charts/pull/682
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to https://github.com/kedacore/charts/pull/682

### CODE REVIEW

File: config/rbac/role.yaml
Comment (legal90) [inline config/rbac/role.yaml:52]: To maintainers: Please note that this overly permissive rule "allow to _get_ any resource" was removed without any replacement. 

Should I keep it untouched, or is it safe to delete it?

Comment (JorTurFer) [reply config/rbac/role.yaml:52]: In helm, we remove it based on configs, but it's required using raw manifests

Comment (legal90) [reply config/rbac/role.yaml]: Thank you, @JorTurFer! I reverted the deletion of that particular rule.

File: controllers/keda/scaledobject_controller.go
Comment (JorTurFer) [inline controllers/keda/scaledobject_controller.go:62]: We have to revert this change because KEDA must allow scaling any resource as default using raw manifests

**Review by JorTurFer — COMMENTED** [review]
Looking good, only 1 comment inline
PTAL @wozniakjan

File: controllers/keda/scaledobject_controller.go
Comment (legal90) [reply controllers/keda/scaledobject_controller.go]: Thank you for the review, @JorTurFer !
I rolled back this line and re-generated `role.yaml` (by `make manifests`). 
Could you please check again?

File: config/rbac/role.yaml
Comment (wozniakjan) [inline config/rbac/role.yaml:22]: I wasn't sure about this being sufficient but it should be fine. For any other reviewers, here is the docs
https://book.kubebuilder.io/reference/raising-events#granting-the-required-permissions

**Review by wozniakjan — APPROVED** [review]
lgtm, assuming e2e tests pass

**Comment by JorTurFer** [comment]
@legal90 , could you fix DCO? it was broken during last commit -> https://github.com/kedacore/keda/pull/6129/checks?check_run_id=32323819193

**Comment by legal90** [comment]
@JorTurFer Yes sure!  I fixed the DCO and rebased the branch now.

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**
